src/Functions/seriesPeriodDetectFFT.cpp

```diff
@@ -0,0 +1,169 @@
+#include "config.h"
+
+#if USE_POCKETFFT
+#    ifdef __clang__
+#        pragma clang diagnostic push
+#        pragma clang diagnostic ignored "-Wshadow"
+#        pragma clang diagnostic ignored "-Wextra-semi-stmt"
+#        pragma clang diagnostic ignored "-Wzero-as-null-pointer-constant"
+#    endif
+
+#    include <pocketfft_hdronly.h>
+
+#    ifdef __clang__
+#        pragma clang diagnostic pop
+#    endif
+
+#    include <cmath>
+#    include <Columns/ColumnArray.h>
+#    include <Columns/ColumnsNumber.h>
+#    include <DataTypes/DataTypeArray.h>
+#    include <DataTypes/DataTypesNumber.h>
+#    include <Functions/FunctionFactory.h>
+#    include <Functions/FunctionHelpers.h>
+#    include <Functions/IFunction.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int BAD_ARGUMENTS;
+extern const int ILLEGAL_COLUMN;
+}
+
+/*Detect Period in time series data using FFT.
+ * FFT - Fast Fourier transform (https://en.wikipedia.org/wiki/Fast_Fourier_transform)
+ * 1. Convert time series data to frequency domain using FFT.
+ * 2. Remove the 0th(the Dc component) and n/2th the Nyquist frequency
+ * 3. Find the peak value (highest) for dominant frequency component.
+ * 4. Inverse of the dominant frequency component is the period.
+*/
+
+class FunctionSeriesPeriodDetectFFT : public IFunction
```

Comment:
Needs performance evaluation.

Reply:
Ran some tests with large numbers : https://pastila.nl/?00e07f72/2751cbfa358751a9d5f4d8485dd3c392#Y6yuvEIwVq4/Dm+0hhpgug==

Please let me know if we need performance evaluation against some specific benchmarks etc.